sklearn/preprocessing/_polynomial.py

```diff
@@ -280,6 +293,12 @@ def fit(self, X, y=None):
         # Instead, we reuse the distance of the 2 fist/last knots.
         dist_min = base_knots[1] - base_knots[0]
         dist_max = base_knots[-1] - base_knots[-2]
+
+        # For periodic splines the spacing of the first / last degree knots
+        # needs to be equal
+        if self.extrapolation == "periodic":
+            dist_min = dist_max = (dist_min + dist_max) / 2
```

Comment:
Maybe, this is the reason that the continuity does fail.
The knots "|" for non-periodic spline of degree 2 looks like
|a|b|1|2|3|4|c|d|
1-4 is the base interval, abcd the intervals for the knots beyond the base inverval.
For a periodic spline this should look like
|3|4|1|2|3|4|
The last 2 splines are removed. The first 2 knots must match the end of the period=|3|4|. |3| and |4| not only means the same knot distance, but also the same coefficient, i.e. the same spline.

What do you think?

Reply:
This did the trick. Thanks!